File: webshop-spring/src/main/java/org/nicolai/webshop/auth/DatabaseApplicationUserDaoService.java
package org.nicolai.webshop.auth;

import org.nicolai.webshop.model.Admin;
import org.nicolai.webshop.model.User;
import org.nicolai.webshop.repository.AdminRepository;
import org.nicolai.webshop.repository.UserRepository;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.stereotype.Repository;

import java.util.Optional;

@Repository("db")
public class DatabaseApplicationUserDaoService implements ApplicationUserDao{

    @Autowired
    AdminRepository adminRepository;

    @Override
    public Optional<ApplicationUser> selectApplicationUserByUsername(String username) {

        Admin user = null;
        try {
            user = adminRepository.findByUsername(username);
            System.out.println(user.getRole().getGrantedAuthorities());
        } catch (Exception e) {
            System.out.println("No user found");
        }
        if (user == null) return Optional.empty();
        ApplicationUser applicationUser = new ApplicationUser(username, user.getPassword(),
                user.getRole().getGrantedAuthorities(), true,
                true ,true, true);
        return Optional.of(applicationUser);
    }
}

File: bank-spring/src/main/kotlin/org/nicolai/bank/auth/DatabaseApplicationUserDaoService.kt
package org.nicolai.bank.auth

import org.springframework.jdbc.core.JdbcTemplate
import org.springframework.stereotype.Repository
import java.util.*

@Repository("db")
class DatabaseApplicationUserDaoService(val db: JdbcTemplate) : ApplicationUserDao {


    override fun selectApplicationUserByUsername(username: String?): Optional<ApplicationUser> {
        /*
        var user: User? = null
        try {
            user = userRepository.findByUsername(username)
        } catch (e: Exception) {
            println("No user found")
        }
        if (user == null) return Optional.empty()
        val applicationUser = ApplicationUser(
            username!!, user.getPassword(),
            null, true,
            true, true, true
        )
        return Optional.of(applicationUser)
        */
         return Optional.empty();
    }
}
